packages/core/src/command/hl7-notification/hl7-notification-webhook-sender-direct.ts

```diff
@@ -121,7 +150,103 @@ export class Hl7NotificationWebhookSenderDirect implements Hl7NotificationWebhoo
           },
         })
     );
-    log(`Done. API notified...`);
+
+    log(`Done.`);
+  }
+
+  private async writeTcmEncounterToDb(
+    tcmEncounterPayload: {
+      id: string;
+      cxId: string;
+      patientId: string;
+      class: string | undefined;
+      facilityName: string | undefined;
+      admitTime: string | undefined;
+      dischargeTime: string | undefined;
+      clinicalInformation: Record<string, unknown>;
+    },
+    triggerEvent: "A01" | "A03"
+  ) {
+    const encounterId = tcmEncounterPayload.id;
+    const admitTime = tcmEncounterPayload.admitTime;
+    const dischargeTime = tcmEncounterPayload.dischargeTime;
+    const { log } = out(
+      `writeTcmEncounterToDb, cx: ${tcmEncounterPayload.cxId}, pt: ${tcmEncounterPayload.patientId}, enc: ${encounterId}`
+    );
+
+    /**
+     * TODO:
+     * We frequently receive "duplicate" ADTs - different messages with the same data for the same event.
+     * We need to deduplicate them to avoid noisy creation errors.
+     */
+    if (triggerEvent === "A01") {
+      log(`POST /internal/tcm/encounter/ A01`);
+      await executeWithNetworkRetries(
+        async () =>
+          axios.post(`${this.apiUrl}/internal/tcm/encounter/`, {
+            ...tcmEncounterPayload,
+            id: encounterId,
+            latestEvent: "Admitted",
+          }),
+        {
+          log,
+        }
+      );
+    } else if (triggerEvent === "A03") {
```

Comment:
Why the conditional, since we're calling the same endpoint w/ the same data, just changing the `latestEvent`?

Reply:
wanted a certain product experience but i think it doesnt matter, updated.